apis/placement/v1alpha1/stagedupdate_types.go

```diff
@@ -0,0 +1,441 @@
+	// in the update run "updateRun1" is "myApproval-updateRun1-stage1".
+	// We will generate the name of the approval request by default if the prefix is absent.
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:validation:Pattern="[A-Za-z0-9]+$"
+	// +kubebuilder:validation:MaxLength=32
+	ApprovalRequestPrefix string `json:"approvalTask,omitempty"`
+
+	// The time to wait after all the clusters in the current stage complete the update before we move to the next stage.
+	// +kubebuilder:default="1h"
+	// +kubebuilder:validation:Pattern="^0|([0-9]+(\\.[0-9]+)?(s|m|h))+$"
+	// +kubebuilder:validation:Type=string
+	// +kubebuilder:validation:Optional
+	WaitTime metav1.Duration `json:"waitTime,omitempty"`
+}
+
+// StagedUpdateRunStatus defines the observed state of the StagedUpdateRun.
+type StagedUpdateRunStatus struct {
+	// UpdateRunStrategy is the collection of the stageConfig in the StagedUpdateStrategy that we are going to use for the update.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MaxItems=128
+	UpdateRunStrategy []StageConfig `json:"updateRunStrategy"`
+
+	// ApplyStrategy is the apply strategy that the stagedUpdateRun is using.
+	// It is the same as the apply strategy in the CRP when we first start the staged update run.
+	// We will NOT update the apply strategy during the update run even if the apply strategy changes in the CRP.
+	// +kubebuilder:validation:Required
+	ApplyStrategy v1beta1.ApplyStrategy `json:"appliedStrategy"`
+
+	// ResourceOverrideSnapshots is a list of ResourceOverride snapshots associated with the selected resources.
+	// The list computed as the beginning of the update run and not updated during the update run.
+	// The list is empty if there are no resource overrides associated with the selected resources.
+	// +kubebuilder:validation:Optional
+	ResourceOverrideSnapshots []v1beta1.NamespacedName `json:"resourceOverrideSnapshots,omitempty"`
+
+	// ClusterResourceOverrides contains a list of applicable ClusterResourceOverride snapshot names associated with the
+	// selected resources.
+	// The list computed as the beginning of the update run and not updated during the update run.
+	// The list is empty if there are no cluster overrides associated with the selected resources.
+	// +kubebuilder:validation:Optional
+	ClusterResourceOverrideSnapshots []string `json:"clusterResourceOverrideSnapshots,omitempty"`
+
+	// CurrentStageIndex is the name of the stage that is in the middle of the update.
+	// If the update run is not started, the value is an empty string.
+	// If we are in the DeletionStage, the name is called "deletionStage".
+	// +kubebuilder:validation:Optional
+	CurrentUpdatingStage string `json:"currentUpdatingStage,omitempty"`
+
+	// Stages list the current status of each stage.
+	// The list is empty if the update run is not started.
+	// +kubebuilder:validation:Optional
+	StagesStatus []StageStatus `json:"stagesStatus,omitempty"`
```

Comment:
Just "Stages" as everything is part of "XXXStatus {}"?

Sometimes you use Foos []FooStatus, other times you use FoosStatus []FooStatus. Be consistent?

See below:
Clusters []ClusterUpdateStatus